services/api/db/serializers.py

```diff
@@ -0,0 +1,132 @@
+                 style={'input_type' : 'password'}, write_only=True)
+    class Meta:
+        model = User
+        fields = ['email', 'password']
+
+class UserProfileSerialiser(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+    #New password type could be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
+    confirm_password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
+    class Meta:
+        fields = ['password', 'confirm_password']
+    def validate(self, attrs):
+        password, confirm_password = attrs.get('password'), attrs.get('confirm_password')
+        if not password and not confirm_password:
+            raise ValueError("Both password and confirm_password are empty")
+        if validate_confirm_password(password,confirm_password):
+            return attrs
+        else:
+            raise serializers.ValidationError("Password and confirm password don't match.")
+  
+class SendPasswordResetEmailSerializer(serializers.Serializer):
+    email = serializers.EmailField(
+        max_length=255,
+        style={'input_type': 'email', 'placeholder': 'Email Address'}
+    )
+    class Meta:
+        fields = ['email']
+    def validate(self, attrs):
+        """
+        Raises: ValidationError: If user is not registered, or token generation fails.
+        """
+        email = attrs.get('email')
+        if not email:
+            raise ValueError("Email field is required")
+        try:
+            user = User.objects.get(email=email)
+        except User.DoesNotExist:
+            # Handle unregistered email in a user-friendly way
+            raise ValidationError('No user with that email address exists. Please register first.')
+        # Handle potential exceptions during encoding and token generation
+        try:
+            uid = urlsafe_base64_encode(force_bytes(user.id))
+            token = PasswordResetTokenGenerator().make_token(user)
+        except Exception as e:
+            raise serializers.ValidationError(f'Failed to generate reset link: {e}')
+        # TODO: Please specify the intended platform or service where you wish to reset your password.
+        link = f"http://localhost:3000/api/user/reset/{uid}/{token}"
+        return attrs
+
+class UserPasswordResetSerializer(serializers.Serializer):
+    password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    confirm_password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    class Meta:
+        fields = ['password', 'confirm_password']       
+    def validate(self, attrs):
+        password, confirm_password = attrs.get('password'), attrs.get('confirm_password')
+        if not password and not confirm_password:
```

Comment:
```suggestion
        if not password and not confirm_password:
```
should or be used instead of and ?

Reply:
changed